src/sentry/api/endpoints/organization_details.py

```diff
@@ -954,7 +967,45 @@ def put(self, request: Request, organization) -> Response:
             with transaction.atomic(router.db_for_write(Organization)):
                 organization, changed_data = serializer.save()
 
-            if "targetSampleRate" in changed_data:
+            if (
+                organization.get_option("sentry:sampling_mode")
+                == DynamicSamplingMode.ORGANIZATION.value
+            ) and "targetSampleRate" in changed_data:
+                with transaction.atomic(router.db_for_write(ProjectOption)):
+                    organization.update_option(
+                        "sentry:target_sample_rate", request.data["targetSampleRate"]
+                    )
+
+            if "samplingMode" in changed_data and request.access.has_scope("org:write"):
+                if (
+                    organization.get_option("sentry:sampling_mode")
+                    == DynamicSamplingMode.PROJECT.value
+                ):
+                    with transaction.atomic(router.db_for_write(ProjectOption)):
+                        for project in organization.project_set.all():
+                            current_rate, is_fallback = get_boost_low_volume_projects_sample_rate(
+                                org_id=organization.id,
+                                project_id=project.id,
+                                error_sample_rate_fallback=None,
+                            )
+                            if current_rate:
+                                project.update_option("sentry:sampling_rate", current_rate)
+
+                        organization.delete_option("sentry:target_sample_rate")
+
+                if (
+                    organization.get_option("sentry:sampling_mode")
+                    == DynamicSamplingMode.ORGANIZATION.value
+                ):
+                    with transaction.atomic(router.db_for_write(ProjectOption)):
+                        if "targetSampleRate" in changed_data:
+                            organization.update_option(
+                                "sentry:target_sample_rate", request.data["targetSampleRate"]
```

Comment:
We should use validated data from serializer rather than `request.data`, sometimes the value is already sanitized (if needed) in a serializer, and we are sure that it passed the validation.

Here it would be `serializer.validated_data["targetSampleRate"] instead of `request.data["targetSampleRate"]`

Reply:
changed